diff --git a/src/ngx_http_upstream_jdomain_module.c b/src/ngx_http_upstream_jdomain_module.c
index 104f02f..94f21d3 100644
--- a/src/ngx_http_upstream_jdomain_module.c
+++ b/src/ngx_http_upstream_jdomain_module.c
@@ -6,7 +6,7 @@
 #define NGX_JDOMAIN_STATUS_DONE 0
 #define NGX_JDOMAIN_STATUS_WAIT 1
 
-#define NGX_JDOMAIN_DEFAULT_SERVER_FAIL_TIMEOUT 10
+#define NGX_JDOMAIN_DEFAULT_SERVER_FAIL_TIMEOUT 30
 #define NGX_JDOMAIN_DEFAULT_SERVER_MAX_FAILS 1
 #define NGX_JDOMAIN_DEFAULT_SERVER_WEIGHT 1
 #define NGX_JDOMAIN_DEFAULT_PORT 80
 